File: ch11/MyNotesPOOO/BusinessLogicLayer/NoteBL.m
//
//  NoteBL.m
//  MyNotes
//
//  Created by MyZiyue on 2018/12/29.
//  Copyright © 2018 MyZiyue.com. All rights reserved.
//

#import "NoteBL.h"

@implementation NoteBL

-(NSMutableArray*)createNote:(Note*)model{
    NoteDAO* dao = [NoteDAO sharedInstance];
    [dao create:model];
    return [dao findAll];
}

-(NSMutableArray*)remove:(Note*)model{
    NoteDAO* dao = [NoteDAO sharedInstance];
    [dao remove:model];
    return [dao findAll];
}

-(NSMutableArray*)findAll{
    NoteDAO* dao = [NoteDAO sharedInstance];
    return [dao findAll];
}

@end

File: ch11/SMyNotes/BusinessLogicLayer/NoteBL.swift
//
//  NoteBL.swift
//  SMyNotes
//
//  Created by MyZiyue on 2018/12/29.
//  Copyright © 2018 MyZiyue.com. All rights reserved.
//

import Foundation

class NoteBL {
    func createNote(model: Note) -> NSMutableArray {
        let dao: NoteDAO = NoteDAO.sharedInstance
        dao.create(model: model)
        return dao.findAll()
    }
    
    func remove(model: Note) -> NSMutableArray {
        let dao: NoteDAO = NoteDAO.sharedInstance
        dao.remove(model: model)
        return dao.findAll()
    }
    
    func findAll() -> NSMutableArray {
        let dao: NoteDAO = NoteDAO.sharedInstance
        return dao.findAll()
    }
}
